### ISSUE TITLE
remove a visitor/IP address from persistent database

### ISSUE THREAD

**Issue by nodiscc** [allinurl/goaccess#2599]
Hi,

How can I remove a visitor/IP address from goaccess' persistent database?

I run goaccess with persistence, and update the HTML report using a systemd timer every hour. [[1]](https://github.com/nodiscc/xsrv/tree/master/roles/monitoring_goaccess) [[2]](https://github.com/nodiscc/xsrv/blob/master/roles/monitoring_goaccess/templates/etc_goaccess_goaccess.conf.j2) [[3]](https://github.com/nodiscc/xsrv/blob/master/roles/monitoring_goaccess/templates/etc_systemd_system_goaccess-update.service.j2), the relevant options are

```
# goaccess.conf
db-path /var/lib/goaccess/db
persist true
restore true
```

```bash
# systemd timer
ExecStart=/bin/bash -c "ls -tr /var/log/apache2/access.log* | xargs zcat -f my.example.org | awk '$8=$1$8' | goaccess - --output /var/www/stats.example.org/index.html"
```

Every once in a while, I will notice a visitor in the `Visitor Hostnames and IPs` who is obviously abusing the service (huge tx amount), and proceed to ban them. After that, I would like to remove this visitor from the table (possibly other tables, but this is the only table I really care about), so that I can focus on possible other offenders without the table being "polluted" by visitors I already banned.

I don't know much about the database format used by goaccess and would accept a quick-and-dirty script a long as it does the job.

Ideally, goaccess could provide a subcommand or option to remove a visitor for the db.

**Comment by allinurl** [comment]
Give it a try by passing `-e`, like so: `-e 192.168.0.1`

**Comment by nodiscc** [comment]
@allinurl this doesn't seem to work, at least when using `exclude-ip` in the configuration file

```diff
 # Exclude an IPv4 or IPv6 from being counted.
 # Ranges can be included as well using a dash in between
 # the IPs (start-end).
 #
 #exclude-ip 127.0.0.1
 #exclude-ip 192.168.0.1-192.168.0.100
 #exclude-ip ::1
 #exclude-ip 0:0:0:0:0:ffff:808:804-0:0:0:0:0:ffff:808:808
 exclude-ip 127.0.0.1
+exclude-ip 73.14x.xxx.xxx
```

And then waiting for the hourly update of the HTML report, the offending IP still appears in the report:

![image](https://github.com/allinurl/goaccess/assets/3950390/5f8897fe-9cb6-4464-b5ae-10e3e751bf2f)

**Comment by allinurl** [comment]
I see. Given your use of persist and restore, the exclude-ip option isn't effective because it operates during parsing, not during the restoration from disk. The challenge is that the counts are already computed, so even if an option to exclude them during restoration is added, it wouldn't accurately reflect the counts. I believe the most suitable solution would be addressed through #117, which is currently in progress. 

Closing this for now but [**stay tuned**](https://twitter.com/goaccess?lang=en) for the release of #117.
